### PR TITLE
feat: call contract method with mutiple params

### PR DESCRIPTION
**Pull request by luizstacio**
### Summary

- Fix encode for call a contract with multiple params;
- Add complementary tests on package `provider`
- Remove `isStruct` from the encoded interface

Contract example;
```
abi TestContract {
  fn foo(value: u64) -> u64;
  fn boo(value: TestStruct) -> TestStruct;
  fn barfoo(value: u64) -> u64;
  fn foobar(value: ()) -> u64;
  fn foobar_no_params() -> u64;
  fn sum(a: u64, b: u64) -> u64;
  fn sum_test(a: u64, test: SumStruct) -> u64;
  fn sum_single(test: SumStruct) -> u64;
  fn sum_multparams(a: u64, b: u64, c: u64, d: u64, e: u64) -> u64;
  fn echo_b256(a: b256) -> b256;
  fn add_ten(param: SingleParamStruct) -> u64;
}
```

### CODE REVIEW

File: packages/providers/src/provider.ts [hunk 1]
```diff
@@ -359,11 +359,13 @@ export default class Provider {
...
-    const isStructArg = dataArray.slice(8, 16).some((b) => b === 0x01);
-    const arg = dataArray.slice(16);
+    // const isStructArg = dataArray.slice(8, 16).some((b) => b === 0x01);
+    // const arg = dataArray.slice(16);
+    const arg = dataArray.slice(8);
 
     let scriptData;
-    if (isStructArg) {
+    // If arguments has more than 8 bites it should be encode with arg offset
+    if (arg.length > 8) {
```
Comment (luizstacio) [inline packages/providers/src/provider.ts:368]: 👆

**Review by QuinnLee — APPROVED** [review]

File: packages/abi-coder/src/interface.ts [hunk 2]
```diff
@@ -110,23 +109,17 @@ export default class Interface {
...
-    return hexlify(
-      concat([
-        selector,
-        new BooleanCoder('isStruct').encode(fragment.inputs[0].type.startsWith('struct')),
-        args,
-      ])
-    );
+    // const isStruct = inputs.some((i) => i.type.startsWith('struct')) || inputs.values.length > 1;
+    // return hexlify(concat([selector, new BooleanCoder('isStruct').encode(isStruct), args]));
+
```
Comment (adlerjohn) [inline packages/abi-coder/src/interface.ts:120]:
```suggestion
```

File: packages/providers/src/provider.ts [hunk 3]
```diff
@@ -359,11 +359,13 @@ export default class Provider {
   }> {
     const dataArray = arrayify(data);
     const functionSelector = dataArray.slice(0, 8);
-    const isStructArg = dataArray.slice(8, 16).some((b) => b === 0x01);
-    const arg = dataArray.slice(16);
+    // const isStructArg = dataArray.slice(8, 16).some((b) => b === 0x01);
+    // const arg = dataArray.slice(16);
+    const arg = dataArray.slice(8);
```
Comment (adlerjohn) [inline packages/providers/src/provider.ts:364]:
```suggestion
    const arg = dataArray.slice(8);
```

File: packages/providers/src/provider.ts [hunk 4]
```diff
@@ -359,11 +359,13 @@ export default class Provider {
...
 
     let scriptData;
-    if (isStructArg) {
+    // If arguments has more than 8 bytes it should be encode with arg offset
```
Comment (adlerjohn) [inline packages/providers/src/provider.ts:367]:
```suggestion
    // If arguments has more than 8 bytes it should be encoded with arg offset
```

**Review by adlerjohn — CHANGES_REQUESTED** [review]

**Review by QuinnLee — APPROVED** [review]
lgtm

Can you add an issue to add more `struct` tests on the abi? I think we may need to update/add more

**Review by AlicanC — APPROVED** [review]

**Review by adlerjohn — APPROVED** [review]

**Review by digorithm — APPROVED** [review]

**Pull request merged**